components/public-api-server/pkg/oidc/service.go

```diff
@@ -28,8 +27,9 @@ import (
 )
 
 type Service struct {
-	dbConn *gorm.DB
-	cipher db.Cipher
+	dbConn   *gorm.DB
+	cipher   db.Cipher
+	stateJWT *StateJWT
```

Comment:
In some way, the fact that it's a JWT is an implementation detail. And as such, here we could go with an interface rather than the fact that it is JWT

You could express this as 
```
type State interface {
  Encode(..)
  Decode(..)
}
```

While this creates indirection, it does help future readers to not have to know the details of how it's encoded, just that it is.

Reply:
Up to you.